File: app/src/main/java/com/savvasdalkitsis/gameframe/grid/model/Grid.java
package com.savvasdalkitsis.gameframe.grid.model;

import android.support.annotation.ColorInt;

import com.savvasdalkitsis.gameframe.model.Moment;

public interface Grid extends Moment<Grid> {

    void setColor(@ColorInt int color, int column, int row);

    Grid fill(int color);

    @ColorInt
    int getColor(int column, int row);
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/grid/model/Grid.kt
package com.savvasdalkitsis.gameframe.grid.model

import android.support.annotation.ColorInt

import com.savvasdalkitsis.gameframe.model.Moment

interface Grid : Moment<Grid> {

    fun setColor(@ColorInt color: Int, column: Int, row: Int)

    fun fill(color: Int): Grid

    @ColorInt
    fun getColor(column: Int, row: Int): Int
}
